sql/logictest: TestLogic failed

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/5558667?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/5558667?buildTab=artifacts#/) on master @ [fc7ae339c3f85ad2a2b28643e47d7c20768fe237](https://github.com/cockroachdb/cockroach/commits/fc7ae339c3f85ad2a2b28643e47d7c20768fe237):


```
[05:10:55] DELETE FROM ab WHERE true;
INSERT INTO ab VALUES (1,1), (2,1), (3,3), (4, 7);
[05:10:55] SELECT b FROM ab GROUP BY a;
[05:10:55] 	-- OK;
[05:10:55] SELECT a+b, count(*) FROM ab JOIN tab ON b=col2 GROUP BY a;
[05:10:55] 	-- OK;
[05:10:55] SELECT a, col1, b+col2, count(*) FROM ab JOIN tab ON b=col2 GROUP BY a, col1;
[05:10:55] 	-- OK;
[05:10:55] SELECT a, b, count(*), count(col2) FROM ab LEFT JOIN tab ON b=col2 GROUP BY a;
[05:10:55] 	-- OK;
[05:10:55] SELECT a, b, count(*) FROM ab RIGHT JOIN tab ON b=col2 GROUP BY a;
[05:10:55] 	-- OK;
[05:10:55] CREATE TABLE xyz (
  x INT PRIMARY KEY,
  y INT,
  z INT,
  INDEX yz (y, z)
);
[05:10:55] rewrote:
CREATE TABLE xyz (x INT8 PRIMARY KEY, y INT8, z INT8, INDEX yz (y, z), FAMILY (z), FAMILY (x, y));


[05:10:55] INSERT INTO xyz VALUES (1, 2, 3), (2, 2, 7), (3, 2, 1), (4, 2, NULL), (5, 3, -1);
[05:10:55] SELECT min(z) FROM xyz WHERE y = 2 GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT min(z) FROM xyz WHERE y = 2 AND z IS NOT NULL GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT min(z) FROM xyz WHERE y = 2 AND z IS NULL GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT min(z) FROM xyz WHERE y = 100 AND z IS NULL GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT max(z) FROM xyz WHERE y = 2 GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT max(z) FROM xyz WHERE y = 2 AND z IS NOT NULL GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT max(z) FROM xyz WHERE y = 2 AND z IS NULL GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] SELECT max(z) FROM xyz WHERE y = 100 GROUP BY y;
[05:10:55] 	-- OK;
[05:10:55] DROP TABLE xyz;
[05:10:56] CREATE TABLE t44469_a (a INT, INDEX (a));
[05:10:56] CREATE TABLE t44469_b (b INT, INDEX (b));
[05:10:56] CREATE TABLE t44469_cd (c INT, d INT, INDEX (c, d));;
[05:10:56] rewrote:
CREATE TABLE t44469_cd (c INT8, d INT8, INDEX (c, d), FAMILY (c, d));


[05:10:56] SELECT DISTINCT ON (b) b
FROM t44469_a INNER LOOKUP JOIN t44469_b ON a = b INNER LOOKUP JOIN t44469_cd ON c = 1 AND d = a
ORDER BY b;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82978 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-queries branch-release-21.2]
- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-17025

This fails only on `SQL Logic Test High VModule Nightly (Bazel)` CI config with
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10127eb]

goroutine 1033 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc003604000, {0x63e29f0, 0xc0005b6280}, {0x5637dc0, 0x858d520})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1053 +0x3d4
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1()
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:755 +0x6b
panic({0x5637dc0, 0x858d520})
	GOROOT/src/runtime/panic.go:1047 +0x266
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x50c
panic({0x5637dc0, 0x858d520})
	GOROOT/src/runtime/panic.go:1047 +0x266
github.com/cockroachdb/cockroach/pkg/util/log.getIdentificationPayload({0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/util/log/server_ident.go:57 +0x6b
github.com/cockroachdb/cockroach/pkg/util/log.makeEntry({0x0, 0x0}, _, _, _)
	github.com/cockroachdb/cockroach/pkg/util/log/log_entry.go:193 +0x88
github.com/cockroachdb/cockroach/pkg/util/log.makeUnstructuredEntry({_, _}, _, _, _, _, {_, _}, {0xc001312b20, 0x3, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/log_entry.go:232 +0xc5
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x0, 0x0}, 0x3, 0x1, 0x0, 0x0, {0x5c72c6a, 0x2a}, {0xc001312b20, 0x3, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:98 +0x5dc
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth({0x0, 0x0}, 0x2, 0x1, 0x0, {0x5c72c6a, 0x2a}, {0xc001312b20, 0x3, 0x3})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39 +0x9d
github.com/cockroachdb/cockroach/pkg/util/log.InfofDepth({0x0, 0x0}, 0x1, {0x5c72c6a, 0x2a}, {0xc001312b20, 0x3, 0x3})
	github.com/cockroachdb/cockroach/bazel-out/k8-dbg/bin/pkg/util/log/log_channels_generated.go:332 +0x87
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Start(0xc0048f7680, {0x0, 0x0}, 0xc0048f6a80, {0x0, 0x0, 0x0, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:365 +0x3e5
github.com/cockroachdb/cockroach/pkg/sql/execinfra.NewMonitor({0x0, 0x0}, 0xc0048f6a80, {0x5befab1, 0xd})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:957 +0x85
github.com/cockroachdb/cockroach/pkg/sql/rowexec.newJoinReader(0xc004ea72c0, 0x1, 0xc0040b2000, {0x6408878, 0xc004ea7e00}, 0xc00355daf8, {0x0, 0x0}, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/rowexec/joinreader.go:478 +0x2032
github.com/cockroachdb/cockroach/pkg/sql/rowexec.NewProcessor({0x63e29f0, 0xc004adc680}, 0xc004ea72c0, 0x1, 0xc00355d9f0, 0xc00355daf8, {0xc003a654f0, 0x1, 0x1}, {0xc003a65510, ...}, ...)
...
```

--- a/pkg/sql/rowexec/joinreader.go
+++ b/pkg/sql/rowexec/joinreader.go
@@ -478,7 +478,7 @@ func newJoinReader(
 		var diskBuffer kvstreamer.ResultDiskBuffer
 		if jr.maintainOrdering {
 			jr.streamerInfo.diskMonitor = execinfra.NewMonitor(
-				jr.Ctx, jr.FlowCtx.DiskMonitor, "streamer-disk", /* name */
+				flowCtx.EvalCtx.Ctx(), jr.FlowCtx.DiskMonitor, "streamer-disk", /* name */
 			)
 			diskBuffer = rowcontainer.NewKVStreamerResultDiskBuffer(
 				jr.FlowCtx.Cfg.TempStorage, jr.streamerInfo.diskMonitor,

